### ISSUE TITLE
Handle remaining FPC warnings/notes

### ISSUE THREAD

**Issue by PMeira** [dss-extensions/dss_capi#82]
With the migration to Free Pascal 3.2, the number of warnings and notes increased. Many notes about inline functions could be simply silenced:

``` 
Ucomplex.pas(214,59) Note: Call to subroutine "function topolar(const a:Double;const b:Double):<record type>;" marked as inline is not inlined
```

**Comment by PMeira** [comment]
I enabled the remaining warnings from the FPC config files and checked most of the messages. That warnings config was inherited from the official repo config for Lazarus/FPC:

https://github.com/dss-extensions/dss_capi/blob/dbefda0203c0039d158658e571e9cc6ab7b31384/src/windows.cfg#L22

Most of the warnings are not useful, but most of them are easy to handle. I handled most of the warnings on the code at API level and the rest, at engine level, can be handled later. We should keep the warnings enabled since they can be helpful. #83 was found in the warnings.
